### PR TITLE
bpo-38312: Add curses.{get,set}_escdelay and curses.{get,set}_tabsize

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue38312

### CODE REVIEW

File: Modules/_cursesmodule.c [hunk 1]
```diff
@@ -3255,6 +3255,67 @@ _curses_setupterm_impl(PyObject *module, const char *term, int fd)
...
+to distinguish between an individual escape character entered on the
+keyboard from escape sequences sent by cursor and function keys.
+[clinic start generated code]*/
+
+static PyObject *
+_curses_set_escdelay_impl(PyObject *module, int ms)
+/*[clinic end generated code: output=43818efbf7980ac4 input=7796fe19f111e250]*/
+{
+    if (ms <= 0) {
+        PyErr_SetString(PyExc_ValueError, "ms must be > 0");
```
Comment (serhiy-storchaka) [inline Modules/_cursesmodule.c:3292]: Why is it an error?

File: Modules/_cursesmodule.c [hunk 2]
```diff
@@ -3255,6 +3255,67 @@ _curses_setupterm_impl(PyObject *module, const char *term, int fd)
...
+        return NULL;
+    }
+
+    int code = set_escdelay(ms);
+    if (code != ERR) {
```
Comment (serhiy-storchaka) [inline Modules/_cursesmodule.c]: You can use `PyCursesCheckERR()`.

File: Modules/_cursesmodule.c (hunk 1 above)
Comment (asottile) [reply Modules/_cursesmodule.c:3292]: it was triggering segmentation fault in my implementation of curses :S

**Review by serhiy-storchaka — COMMENTED** [review]
Please add tests and a What's New entry.

File: Modules/_cursesmodule.c (hunk 2 above)
Comment (asottile) [reply Modules/_cursesmodule.c]: ah, I copied this from elsewhere in the file -- should I fix those as well?

**Comment by asottile** [comment]
> Please add tests and a What's New entry.

For tests I'm not sure it's possible to demonstrate the behaviour -- should I just call the functions and ensure they don't crash?

File: Modules/_cursesmodule.c (hunk 2 above)
Comment (serhiy-storchaka) [reply Modules/_cursesmodule.c]: It is better to do this in a separate tiny PR (no issue is needed).

File: Modules/_cursesmodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_cursesmodule.c:3292]: Thanks.

**Comment by serhiy-storchaka** [comment]
Yes, testing that functions exist and do not crash is the minimum. Many functions in the curses module do not have better testing.

File: Modules/_cursesmodule.c (hunk 2 above)
Comment (asottile) [reply Modules/_cursesmodule.c]: awesome, done!

**Comment by asottile** [comment]
tests and what's new added \o/ -- thanks for the review :)

**Review by serhiy-storchaka — COMMENTED** [review]
Add also versionadded to get_escdelay and get_tabsize.

Oh, seems there is no get_tabsize in ncurses. But it is okay.

**Comment by asottile** [comment]
> Add also versionadded to get_escdelay and get_tabsize.

oops! added :)

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**